File: Python3/3043_find_the_length_of_the_longest_common_prefix.py
# https://leetcode.cn/problems/find-the-length-of-the-longest-common-prefix/

class Solution:
    def longestCommonPrefix(self, arr1: list[int], arr2: list[int]) -> int:
        s = set()
        for i in range(len(arr1)):
            num = arr1[i]
            while num:
                s.add(num)
                num //= 10
        
        res = 0
        for i in range(len(arr2)):
            num = arr2[i]
            while num and num not in s:
                num //= 10
            res = max(res, num)
        return len(str(res)) if res else 0

File: Cpp/3043_find_the_length_of_the_longest_common_prefix.cpp
// https://leetcode.cn/problems/find-the-length-of-the-longest-common-prefix/

#include <iostream>
#include <vector>
#include <unordered_set>
#include <string>

using namespace std;

class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        unordered_set<int> s;
        for (int num: arr1) {
            while (num > 0) {
                s.insert(num);
                num /= 10;
            }
        }
        int res = 0;
        for (int num: arr2) {
            while (num > 0 && !s.count(num)) {
                num /= 10;
            }
            res = max(res, num);
        }
        return res > 0 ? to_string(res).size() : 0;
    }
};

int main() {
    Solution sol;
    vector<int> arr1 = {1, 10, 100};
    vector<int> arr2 = {1000};
    cout << sol.longestCommonPrefix(arr1, arr2) << '\n';
    return 0;
}